### PR TITLE
Cache HTTP call for a day to avoid multiple calls

### PR DESCRIPTION
**Pull request by mitogh**
**Ticket** https://central.tri.be/issues/99486

Makes sure next calls are based from the transient instead of the HTTP
call.

### CODE REVIEW

File: src/Tribe/Aggregator.php [hunk 1]
```diff
@@ -3,6 +3,14 @@
 defined( 'WPINC' ) or die;
 
 class Tribe__Events__Aggregator {
+	/**
+	 * Cache key used to storage the services list returned by the call to:
+	 * - Tribe__Events__Aggregator__Service::instance()->get_origins();
+	 *
+	 * @since TBD
+	 */
+	const SERVICES_CACHE_KEY = 'tribe_aggregator_services_list';
```
Comment (bordoni) [inline src/Tribe/Aggregator.php]: Avoid new Constants, use a public prop. Access by `tribe( 'tec.aggregator' )->key_cache_services`.
When saving keys let's try to always start with `key_...`

File: src/Tribe/Aggregator.php [hunk 2]
```diff
@@ -582,4 +593,13 @@ public function hook() {
...
 		return true;
 	}
+
+	/**
+	 * Function used to remove cache stored in transients.
+	 *
+	 * @since TBD
+	 */
+	public function clear_services_list_cache() {
+		delete_transient( self::SERVICES_CACHE_KEY );
```
Comment (bordoni) [inline src/Tribe/Aggregator.php]: `return` the result o the function call.

File: src/Tribe/Aggregator/Admin_Bar.php [hunk 3]
```diff
@@ -80,7 +80,13 @@ public function init( WP_Admin_Bar $wp_admin_bar = null ) {
 			return;
 		}
 
-		$service_response = Tribe__Events__Aggregator__Service::instance()->get_origins();
+		$service_response = get_transient( Tribe__Events__Aggregator::SERVICES_CACHE_KEY );
+
+		// Save HTTP request into a transient
+		if ( false === $service_response ) {
+			$service_response = Tribe__Events__Aggregator__Service::instance()->get_origins();
+			set_transient( Tribe__Events__Aggregator::SERVICES_CACHE_KEY, $service_response, DAY_IN_SECONDS );
```
Comment (bordoni) [inline src/Tribe/Aggregator/Admin_Bar.php]: I have a feeling like we have some cache in place for the origins, or we had it in the past and was removed for reason. Or at least we need to invalidate cache once the user adds a new license.

File: src/Tribe/Aggregator.php (hunk 1 above)
Comment (mitogh) [reply src/Tribe/Aggregator.php]: Curios here of the reasons behind this also wondering if this should be the approach from now on as in some places the `CONST` is the current way to do this.

File: src/Tribe/Aggregator/Admin_Bar.php (hunk 3 above)
Comment (mitogh) [reply src/Tribe/Aggregator/Admin_Bar.php]: 👍

File: src/Tribe/Aggregator.php (hunk 2 above)
Comment (mitogh) [reply src/Tribe/Aggregator.php]: 👍

File: src/Tribe/Aggregator/Admin_Bar.php (hunk 3 above)
Comment (mitogh) [reply src/Tribe/Aggregator/Admin_Bar.php]: Based on the history of the file there seems that never has been stored in some sort of caching mechanism  (the response.)

File: src/Tribe/Aggregator.php (hunk 1 above)
Comment (bordoni) [reply src/Tribe/Aggregator.php]: Yes, so the problem is that there is no way to deprecate `CONST` on PHP 5.2 so we are avoiding adding any new ones. I have a ticket out for me to update that on our Engineering Docs.

File: src/Tribe/Aggregator/Admin_Bar.php (hunk 3 above)
Comment (bordoni) [reply src/Tribe/Aggregator/Admin_Bar.php]: Pefecto!

**Pull request merged**